cwms-data-api/src/main/java/cwms/cda/data/dao/StreamDao.java

```diff
@@ -1,130 +1,257 @@
-                    .withStreamLocations(getStreamLocationsOnStream(streamId, unitSystem,
-                            officeId))
-                    .withTributaries(getTributaries(streamId, unitSystem, officeId))
-                    .withStreamReaches(getReaches(streamId, officeId))
+    static List<Stream> buildStreamListFromResultSet(ResultSet result) throws SQLException
+    {
+        List<Stream> retVal = new ArrayList<>();
+        while(result.next())
+        {
+            Stream stream = new Stream.Builder()
+                    .withStartsDownstream(result.getBoolean(STATIONING_STARTS_DS_COLUMN_INDEX))
+                    .withId(new CwmsId.Builder()
+                            .withName(result.getString(STREAM_STREAM_ID_COLUMN_INDEX))
+                            .withOfficeId(result.getString(STREAM_OFFICE_ID_COLUMN_INDEX))
+                            .build())
+                    .withFlowsIntoStreamNode(buildStreamNode(result.getString(STREAM_OFFICE_ID_COLUMN_INDEX),
+                            result.getString(STREAM_FLOWS_INTO_STREAM_COLUMN_INDEX),
+                            result.getDouble(STREAM_FLOWS_INTO_STATION_COLUMN_INDEX),
+                            Bank.fromCode(result.getString(STREAM_FLOWS_INTO_BANK_COLUMN_INDEX)),
+                            DB_STATION_UNITS))
+                    .withDivertsFromStreamNode(buildStreamNode(result.getString(STREAM_OFFICE_ID_COLUMN_INDEX),
+                            result.getString(STREAM_DIVERTS_FROM_STREAM_COLUMN_INDEX),
+                            result.getDouble(STREAM_DIVERTS_FROM_STATION_COLUMN_INDEX),
+                            Bank.fromCode(result.getString(STREAM_DIVERTS_FROM_BANK_COLUMN_INDEX)),
+                            DB_STATION_UNITS))
+                    .withLength(result.getDouble(STREAM_STREAM_LENGTH_COLUMN_INDEX))
+                    .withAverageSlope(result.getDouble(STREAM_AVERAGE_SLOPE_COLUMN_INDEX))
+                    .withComment(result.getString(STREAM_COMMENTS_COLUMN_INDEX))
+                    .withLengthUnits(DB_STATION_UNITS)
+                    .withSlopeUnits(DB_STREAM_SLOPE_UNITS)
                     .build();
+            retVal.add(stream);
+        }
+        return retVal;
+    }
 
+    static StreamNode buildStreamNode(String officeId, String streamId, Double station, Bank bank, String stationUnits) {
+        StreamNode retVal;
+        try {
+            retVal = new StreamNode.Builder()
+                    .withStreamId(new CwmsId.Builder()
+                            .withName(streamId)
+                            .withOfficeId(officeId)
+                            .build())
+                    .withStation(station)
+                    .withBank(bank)
+                    .withStationUnits(stationUnits)
+                    .build();
+            retVal.validate();
+        } catch (FieldException e) {
+            retVal = null; //if missing required fields, don't build a stream node, just return null
```

Comment:
This mean the database is returning an invalid StreamNode? When does that happen? that should get logged at the very least. It should probably even just be a error code 500 returned to the client.

Reply:
yeah, i double checked the db store call and it only stores the station and bank if the divertsFrom/flowsInto stream id is supplied. So there's no cases in practice where you'd get back an invalid streamNode. Going to update this to just build the streamnode conditionally on if the stream id exists for the diversion/flows into stream

Reply:
updated this to just null check the streamId